src/monitors/CallGraphMonitor.v3

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 Wizard Authors. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// A general-purpose profiler.
+def monitor_ = MonitorRegistry.add(
+	"callgraph", "Extract dynamic call graph and output in dot format.",
+	CallGraphMonitor.new());
+def depthOpt = monitor_.options.newIntOption("depth", -1, "limit the maximum call tree depth");
+def filterOpt = monitor_.options.newDeclFilterOption("calls", "apply profiling to selected functions");
+
+class CallGraphMonitor extends Monitor {
```

Comment:
Thanks for writing this monitor! This is cool functionality to have. The main feedback I have is that it collects similar information to both the `calls` monitor and the `profile` monitor. Of those two, maybe it makes sense to merge the dot graph functionality into the latter?